### ISSUE TITLE
PERsim: possible bug in insert sampling

### ISSUE THREAD

**Issue by stuckyb** [imgag/ngs-bits#36]
Hello,
I've been experimenting with PERsim recently, and I was occasionally seeing some weird output reads with read indexes in the reference that didn't match the actual data.  After looking at the source code, I think I might have found the problem.

It's in `main.cpp`, lines 131-135:
```
std::uniform_int_distribution<int> start_dist(reg.start()-is/2, reg.end()-is/2);
int start = start_dist(gen);
int end = start + is;
if (start<0) start=0;
if (end>reference.lengthOf(chr)) end = reference.lengthOf(chr);
```

The issue, I think, is that if a value of `start` is sampled that is < 0 and `is` is sufficiently small, `start + is` can also be < 0.  In that case, the value of `start` is later corrected to 0, but the value of `end` will remain < 0.  Thus, you end up with `start` at the beginning of the reference and `end` at a nonsense position.

I've not had time to properly confirm this or make a pull request, but I thought it might be useful to report this now in case it is actually a bug.

Thank you!

**Comment by marc-sturm** [comment]
Hi,

thanks for reporting this.
Yes, from the code it looks like the case you describe can happen.
I will look into it over the next few days.

Thanks,
  Marc

**Comment by marc-sturm** [comment]
Hi,

I hopefully fixed the problem with d049707c62b195c544536a94a7c50faf57df833a
If you can verify that the problem is fixed, please close this issue.

If you still encounter this problem, please provide the complete command line of the call and the 'roi' file, so that I can start debugging.

Best,
  Marc

**Comment by stuckyb** [comment]
That should do it.  Thanks!
